### PR TITLE
CPM-116: Use DSM Tree on Product grid

### PR DESCRIPTION
**Pull request by pierallard**
This PR replace the jsTree on the product grid by a one using DSM Tree.
This selector (CategoryTrees) is defined by:
- A Tree
- A Tree Switcher (See comment on the switcher file)
- A "include sub categories" boolean.

All the logic of the routes, callbacks is defined "out the .tsx components".
The CategoryTrees can be used elsewhere if needed (and it will for the categories page)

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/CategoryTreeFetcher.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { CategoryTreeModel } from "@akeneo-pim-community/shared/src/components/CategoryTree/CategoryTree";
```
Comment (pierallard) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/CategoryTreeFetcher.ts:1]: It's just a move into a proper file.

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                  }}
+                >
+                  {tree.label}
+                </Dropdown.Item>
+              ))}
+            </Dropdown.ItemCollection>
+          ) : (
+            <div style={{padding: '15px 40px 30px', textAlign: 'center'}}>
+              <ProductCategoryIllustration size={'100%'} />
+              No matches found!
```
Comment (pierallard) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx:99]: miss translation

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datafilter/filter/product_category-filter.js [hunk 3]
```diff
@@ -73,7 +90,8 @@ define(['jquery', 'underscore', 'oro/datafilter/number-filter', 'pim/tree/view',
       });
 
       mediator.on('grid_action_execute:product-grid:delete', function () {
-        TreeView.refresh();
+        // TODO
```
Comment (pierallard) [inline src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datafilter/filter/product_category-filter.js:93]: TODO

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import styled from 'styled-components';
+import {
+  Dropdown,
+  useBooleanState,
+  ArrowDownIcon,
+  ProductCategoryIllustration,
+  TextInput,
+  SearchIcon,
+} from 'akeneo-design-system/lib';
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx:10]:
```suggestion
} from 'akeneo-design-system';
```

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+const __ = require('oro/translator');
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx:11]:
```suggestion
import {useTranslate} from '@akeneo-pim-community/legacy-bridge';
```

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  border-bottom: 1px solid ${getColor('purple100')};
+  cursor: pointer;
+  display: flex;
+  justify-content: space-between;
+  height: 40px;
+  align-items: center;
+`;
+
+const CategoryTreeSwitcherText = styled.span`
+  color: ${getColor('purple100')};
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx:29]:
```suggestion
  color: ${getColor('brand', 100)};
```

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+const __ = require('oro/translator');
+import {getColor} from 'akeneo-design-system';
+import {CategoryTreeRoot} from './CategoryTrees';
+
+const CategoryTreeSwitcherContainer = styled(Dropdown)`
+  width: 100%;
+`;
+
+const CategoryTreeSwitcherButton = styled.div`
+  border-bottom: 1px solid ${getColor('purple100')};
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx:20]:
```suggestion
  border-bottom: 1px solid ${getColor('brand', 100)};
```

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTrees.tsx [hunk 8]
```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import {CategoryTree, CategoryTreeModel} from './CategoryTree';
+import {BooleanInput} from 'akeneo-design-system/lib/components/Input/BooleanInput/BooleanInput';
+import {Tree} from 'akeneo-design-system/lib';
+const __ = require('oro/translator');
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTrees.tsx:5]: `useTranslate` as well?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+                    onClick(tree.id);
+                    close();
+                  }}
+                >
+                  {tree.label}
+                </Dropdown.Item>
+              ))}
+            </Dropdown.ItemCollection>
+          ) : (
+            <div style={{padding: '15px 40px 30px', textAlign: 'center'}}>
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx:97]:
```suggestion
            <div style={{padding: '15px 40px 30px', textAlign: 'center'}}>
```
Why not a styled component here?

File: src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datafilter/filter/product_category-filter.js (hunk 3 above)
Comment (ValentinMumble) [inline src/Oro/Bundle/PimDataGridBundle/Resources/public/js/datafilter/filter/product_category-filter.js:93]: TODO

**Review by ValentinMumble — APPROVED** [review]
😍

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx (hunk 9 above)
Comment (juliensnz) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTreeSwitcher.tsx:97]: Could you please use a styled component here?

**Review by juliensnz — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTrees.tsx [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+import {CategoryTreeSwitcher} from './CategoryTreeSwitcher';
+import styled from 'styled-components';
+import {getColor} from 'akeneo-design-system';
+
+const CategoryTreesContainer = styled.div`
+  height: calc(100vh - 110px);
+  border-bottom: 1px solid ${getColor('grey80')};
```
Comment (nmarniesse) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTrees.tsx:12]:
```suggestion
  border-bottom: 1px solid ${getColor('grey', 80)};
```

**Review by nmarniesse — APPROVED** [review]

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/CategoryTreeFetcher.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
+import {CategoryTreeModel} from '@akeneo-pim-community/shared/src/components/CategoryTree/CategoryTree';
+
+type CategoryResponse = {
+  attr: {
+    id: string;
+    'data-code': string;
+  };
+  children?: CategoryResponse[];
+  data: string;
+  state: string; // 'closed jstree-root' | 'leaf' | 'closed' | 'leaf toselect jstree-checked';
```
[the remark this answers is not in the document]
Comment (pierallard) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/CategoryTreeFetcher.ts:10]: In fact, it comes from backend, and can be a combination of all these... I updated comment

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeView.tsx [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+const Router = require('pim/router');
+const __ = require('oro/translator');
+
+type State = {
+  selectedNode: number;
+  selectedTree: number;
+  includeSub: boolean;
+};
+
+class TreeView {
```
[the remark this answers is not in the document]
Comment (pierallard) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeView.tsx:17]: Discussed IRL!

**Pull request merged**